s3indexer/app.py

```diff
@@ -37,158 +47,197 @@ def dict_factory(cursor, row):
+    #  'form': None,
+    #  'hashes': {},
+    #  'metadata': {},
+    #  'rev': '20589597',
+    #  'size': None,
+    #  'updated_date': '2019-10-16T16:50:42.653553',
+    #  'uploader': 'foo@bar.edu',
+    #  'urls': [],
+    #  'urls_metadata': {},
+    #  'version': None}
+    url = f"{config['url'].rstrip('/')}?urls=[]&start=00000000-0000-0000-0000-000000000000&limit=1024"
+    auth = (config["username"], config["password"])
+    indexd_response = requests.get(url, auth=auth)
+    assert indexd_response.status_code == 200, f'Retrieve index from indexd should return 200 status: {indexd_response.status_code} {indexd_response.text}'
+    response = indexd_response.json()
+    assert 'records' in response, f'Retrieve index from indexd should return records: {indexd_response.status_code} {indexd_response.text}'
+    for record in response['records']:
+        if not record.get('file_name', None):
+            continue
+        yield record
+
+
+def index(args, logger):
+    """Calls indexd, processes blank records."""
+    # sql lite connection
+    conn = get_db_connection(args.state_dir)
 
-    with open('{}/offset.{}.txt'.format(state_dir, bucket_name), 'w'
-              ) as data_file:
-        data_file.write(offset['Key'])
+    bucket = get_data_upload_bucket(args.config_path)
+    indexd_config = get_config_file(args.config_path)
+    # render to string so we can pass to indexer
+    config_file = json.dumps(indexd_config)
+    #
+    indexd_db_creds = get_indexd_db_creds(args.indexd_creds_path)
+    indexd_db = get_indexd_db_connection(indexd_db_creds)
+
+
+    bucket_name = bucket['name']
+    endpoint_url = bucket['credentials'].get('endpoint_url', None)
+    aws_access_key_id = bucket['credentials']['aws_access_key_id']
+    aws_secret_access_key = bucket['credentials']['aws_secret_access_key']
+    aws_region = bucket['region']
+    #for record in get_blank_indexd_records_via_api(indexd_config):
+    for record in get_blank_indexd_records_via_db(indexd_db):
+        input_url = f"s3://{bucket_name}/{record['did']}/{record['file_name']}"
+        processed_url = get_processed(conn, input_url)
+        if expired(processed_url, args.max_attempts):
+            logger.info(f'echo expired {json.dumps(processed_url,default=str)}')
+            continue
+        if pending(processed_url, args.attempt_intervals):
+            logger.info(f'echo pending {json.dumps(processed_url,default=str)}')
+            continue
+        processed_url = increment(processed_url)
+        put_processed(conn, processed_url)
+        endpoint_env_var = None
+        if endpoint_url:
+            endpoint_env_var = 'AWS_ENDPOINT={}'.format(endpoint_url)
+        print("AWS_REGION={} AWS_ACCESS_KEY_ID={} AWS_SECRET_ACCESS_KEY={} CONFIG_FILE='{}' INPUT_URL={} {} {}".format(
+            aws_region,
+            aws_access_key_id,
+            aws_secret_access_key,
+            config_file,
+            input_url,
+            endpoint_env_var,
```

Comment:
```suggestion
            endpoint_env_var or '',
```
Otherwise, it inserts the string 'None' when the endpoint_url is undefined, breaking the output.